pkg/trace/api/container_linux.go

```diff
@@ -71,50 +81,131 @@ func (i *noCgroupsProvider) GetContainerID(_ context.Context, h http.Header) str
 
 // NewIDProvider initializes an IDProvider instance using the provided procRoot to perform cgroups lookups in linux environments.
 func NewIDProvider(procRoot string) IDProvider {
-	reader, err := cgroups.NewReader()
+	// taken from pkg/util/containers/metrics/system.collector_linux.go
+	var hostPrefix string
+	if strings.HasPrefix(procRoot, "/host") {
+		hostPrefix = "/host"
+	}
+
+	reader, err := cgroups.NewReader(
+		cgroups.WithCgroupV1BaseController(cgroupV1BaseController),
+		cgroups.WithProcPath(procRoot),
+		cgroups.WithHostPrefix(hostPrefix),
+		cgroups.WithReaderFilter(cgroups.ContainerFilter), // Will parse the path in /proc/<pid>/cgroup to get the container ID.
+	)
+
 	if err != nil {
 		log.Warnf("Failed to identify cgroups version due to err: %v. APM data may be missing containerIDs for applications running in containers. This will prevent spans from being associated with container tags.", err)
 		return &noCgroupsProvider{}
 	}
 	cgroupController := ""
 	if reader.CgroupVersion() == 1 {
-		cgroupController = "memory" // The 'memory' controller is used by the cgroupv1 utils in the agent to parse the procfs.
+		cgroupController = cgroupV1BaseController // The 'memory' controller is used by the cgroupv1 utils in the agent to parse the procfs.
```

Comment:
comment probably no longer needed?

Reply:
Sorry about that, this PR was based of another that got merged yesterday (https://github.com/DataDog/datadog-agent/pull/21236)